### PR TITLE
Updated docs to include INFO latencystats and LATENCY HISTOGRAM

### PR DESCRIPTION
**Pull request by filipecosta90**
Apart from the INFO latencystats and LATENCY HISTOGRAM additions, there is also the:

@itamarhaber changed a minor typo on Readme also.

### CODE REVIEW

File: commands/info.md [hunk 1]
```diff
@@ -378,6 +379,19 @@ For each command type, the following line is added:
 
 *   `cmdstat_XXX`: `calls=XXX,usec=XXX,usec_per_call=XXX,rejected_calls=XXX,failed_calls=XXX`
 
+The **latencystats** section provides latency percentile distribution statistics based
+ on the command type.
```
Comment (oranagra) [inline commands/info.md:383]: new styling rules in README. short lines, no word wrapping
```suggestion
The **latencystats** section provides latency percentile distribution statistics based on the command type.
```

File: commands/latency-histogram.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      4# (integer) 8 => (integer) 99940
+      5# (integer) 16 => (integer) 99968
+      6# (integer) 33 => (integer) 100000
+```
+
+@return
+
+@map-reply: specifically:
+
+The command returns an map where each key is a command name, and each value is a histogram element.
```
Comment (oranagra) [inline commands/latency-histogram.md:58]: probably worth mentioning the nested map (calls and buckets here)
```suggestion
The command returns a map where each key is a command name, and each value is a histogram element.
```

File: commands/latency-histogram.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+```
+127.0.0.1:6379> LATENCY HISTOGRAM set
+1# "set" => 1# "calls" => (integer) 100000
+   2# "histogram_usec" => 1# (integer) 1 => (integer) 99583
+      2# (integer) 2 => (integer) 99852
+      3# (integer) 4 => (integer) 99914
+      4# (integer) 8 => (integer) 99940
+      5# (integer) 16 => (integer) 99968
+      6# (integer) 33 => (integer) 100000
+```
```
Comment (oranagra) [inline commands/latency-histogram.md:52]: feels a little too verbose to me.
maybe it's enough to just have one RESP version and not both.

**Comment by oranagra** [comment]
@itamarhaber @yoav-steinberg @zuiderkwast we may need to come up with a plan quickly.
while there's one pending PR that updates the commands.json, any other PR that's opened to do the same will have conflicts.
maybe git can automatically solve them (being identical), but i think GH still shows a conflict, which will require an extra manual step.

File: commands/latency-histogram.md (hunk 3 above)
Comment (filipecosta90) [reply commands/latency-histogram.md]: removed the RESP2 variant @oranagra given IMHO we want the most up-to-date state on the docs.

File: commands/latency-histogram.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+@return
+
+@map-reply: specifically:
+
+The command returns a map where each key is a command name, and each value is a histogram object with the total calls, and time buckets inner map.
```
Comment (filipecosta90) [inline commands/latency-histogram.md:36]: @oranagra as discussed addding the extra details of the inner map and histogram object details

File: commands/latency-histogram.md [hunk 5]
```diff
@@ -55,5 +33,5 @@ RESP3 reply
 
 @map-reply: specifically:
 
-The command returns a map where each key is a command name, and each value is a histogram element.
+The command returns a map where each key is a command name, and each value is a histogram object with the total calls, and time buckets inner map.
```
Comment (oranagra) [inline commands/latency-histogram.md:36]:
```suggestion
The command returns a map where each key is a command name, and each value is a map with the total calls, and an inner map of the histogram time buckets.
```

**Review by oranagra — APPROVED** [review]

File: 1.log [hunk 6]
```diff
@@ -0,0 +1,5 @@
+find tmp -name '*.spell' | xargs cat > tmp/spelling-errors
+cat tmp/spelling-errors
+commands/info.md latencystats
+test ! -s tmp/spelling-errors
+makefile:44: recipe for target 'spell' failed
```
Comment (zuiderkwast) [inline 1.log:5]: File added by mistake?

File: commands/latency-histogram.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+This command requires the extended latency monitoring feature to be enabled (by default is enabled).
+If you need to enable it, use 'CONFIG SET latency-tracking yes'.
+
+@examples
+
+```
+127.0.0.1:6379> LATENCY HISTOGRAM set
+1# "set" => 1# "calls" => (integer) 100000
+   2# "histogram_usec" => 1# (integer) 1 => (integer) 99583
+      2# (integer) 2 => (integer) 99852
```
Comment (zuiderkwast) [inline commands/latency-histogram.md:25]: Is this how redis-cli displays nested maps? (Maybe we need to do something about it in a different PR. It looks like it's misaligned in some way.)

**Review by zuiderkwast — COMMENTED** [review]
Looks good, but I didn't look at the details.

Regarding RESP3, we should really do something about #1521 soon, perhaps for 7.0....

Regarding commands.json, we should have some way to fetch it from Redis when the website is updated. Then we can ignore commands.json in the redis-doc repo and later delete it.

File: commands/latency-histogram.md (hunk 7 above)
Comment (oranagra) [reply commands/latency-histogram.md:25]: yes. it's indeed ugly.
thing is that we don't wanna add a newline if the map contains primitive, just when it contains a compound object.
feel free to look into if if you happen to have time.

File: 1.log (hunk 6 above)
Comment (filipecosta90) [reply 1.log]: yes. thank you for catching it :)

**Review by zuiderkwast — APPROVED** [review]

File: commands.json [hunk 8]
```diff
@@ -6985,6 +7001,32 @@
...
     },
+    "LATENCY HISTOGRAM": {
+        "summary": "Return the cumulative distribution of latencies of a subset of commands or all.",
+        "since": "7.0.0",
+        "group": "server",
+        "complexity": "O(N) where N is the number of commands with latency information being retrieved.",
+        "acl_categories": [
+            "@admin",
+            "@slow",
+            "@dangerous"
```
Comment (yoav-steinberg) [inline commands.json:7012]: dangerous? Why?
I know this flag is a mess and it's unclear when to use it. My personal guideline is that a command is _dangerous_ if we want to be prompted by _redis-cli_ "Are you sure you want to run this [y/N]?" after we press ENTER. And by this standard I see no problem with this command.

File: commands/info.md [hunk 9]
```diff
@@ -378,6 +379,18 @@ For each command type, the following line is added:
 
 *   `cmdstat_XXX`: `calls=XXX,usec=XXX,usec_per_call=XXX,rejected_calls=XXX,failed_calls=XXX`
 
+The **latencystats** section provides latency percentile distribution statistics based on the command type.
+
+ By default, the exported latency percentiles are the p50, p99, and p999.
+ If you need to change the exported percentiles, use 'CONFIG SET latency-tracking-info-percentiles "50.0 99.0 99.9"'.
+
+ This section requires the extended latency monitoring feature to be enabled (by default is enabled).
```
Comment (yoav-steinberg) [inline commands/info.md:387]:
```suggestion
 This section requires the extended latency monitoring feature to be enabled (by default it's enabled).
```

File: commands/info.md [hunk 10]
```diff
@@ -378,6 +379,18 @@ For each command type, the following line is added:
...
+ If you need to enable it, use 'CONFIG SET latency-tracking yes'.
```
Comment (yoav-steinberg) [inline commands/info.md:388]:
```suggestion
 If you need to enable it, use `CONFIG SET latency-tracking yes`.
```

File: commands/info.md [hunk 11]
```diff
@@ -378,6 +379,18 @@ For each command type, the following line is added:
 
 *   `cmdstat_XXX`: `calls=XXX,usec=XXX,usec_per_call=XXX,rejected_calls=XXX,failed_calls=XXX`
 
+The **latencystats** section provides latency percentile distribution statistics based on the command type.
+
+ By default, the exported latency percentiles are the p50, p99, and p999.
+ If you need to change the exported percentiles, use 'CONFIG SET latency-tracking-info-percentiles "50.0 99.0 99.9"'.
```
Comment (yoav-steinberg) [inline commands/info.md:385]:
```suggestion
 If you need to change the exported percentiles, use `CONFIG SET latency-tracking-info-percentiles "50.0 99.0 99.9"`.
```

File: commands/info.md [hunk 12]
```diff
@@ -378,6 +379,18 @@ For each command type, the following line is added:
...
+
+ This section requires the extended latency monitoring feature to be enabled (by default is enabled).
+ If you need to enable it, use 'CONFIG SET latency-tracking yes'.
+
+For each command type, the following line is added:
+
+*   `latency_percentiles_usec_XXX`: `p<percentile 1>=<percentile 1 value>,p<percentile 2>=<percentile 2 value>,...`
```
Comment (yoav-steinberg) [inline commands/info.md:392]:
```suggestion
*   `latency_percentiles_usec_XXX: p<percentile 1>=<percentile 1 value>,p<percentile 2>=<percentile 2 value>,...`
```

File: commands/latency-histogram.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
+The `LATENCY HISTOGRAM` command reports a cumulative distribution of latencies in the format of an histogram for each of the specified command names.
```
Comment (yoav-steinberg) [inline commands/latency-histogram.md:1]:
```suggestion
The `LATENCY HISTOGRAM` command reports a cumulative distribution of latencies in the format of a histogram for each of the specified command names. 
```

File: commands/latency-histogram.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+* The total calls for that command.
+* A map of time buckets:
+  * Each bucket represents a latency range.
+  * Each bucket covers twice the previous bucket's range.
+  * Empty buckets are not printed.
+  * The tracked latencies are between 1 microsecond and roughly 1 second.
+  * Everything above 1 sec is considered +Inf.
+  * At max there will be log2(1000000000)=30 buckets.
+
+This command requires the extended latency monitoring feature to be enabled (by default is enabled).
```
Comment (yoav-steinberg) [inline commands/latency-histogram.md:16]:
```suggestion
This command requires the extended latency monitoring feature to be enabled (by default it's enabled).
```

File: commands/latency-histogram.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+If you need to enable it, use 'CONFIG SET latency-tracking yes'.
```
Comment (yoav-steinberg) [inline commands/latency-histogram.md:17]:
```suggestion
If you need to enable it, use `CONFIG SET latency-tracking yes`.
```

**Review by yoav-steinberg — COMMENTED** [review]
LGTM, minor suggestions.

File: commands.json (hunk 8 above)
Comment (oranagra) [reply commands.json:7012]: this acl **category** is automatically added when the `**admin**` flag is set.

the dangerous category is now defined:
> Potentially dangerous commands (each should be considered with care for various reasons).

i.e. any command that could have some reason to be considered dengarous, is flagged, and users needs to enable them one by one if they want it exposed (e.g. FLUSHDB).

File: commands/info.md (hunk 12 above)
Comment (oranagra) [reply commands/info.md:392]: why don't we quote that whole thing? i.e. it's an example of an output.

Comment (filipecosta90) [reply commands/info.md]: @oranagra , with @yoav-steinberg 's suggestion we have it fully quoted now:
https://github.com/redis/redis-doc/pull/1733/files?authenticity_token=3Fc1AxOaZ6mXciW6ysorP5dxHiAh1vl1A-VHBn96YQvGTSnXBOV8lUj_te5XLzysq_kA12LzUKf37ffwPZotWg&file-filters%5B%5D=.md&file-filters%5B%5D=No+extension#diff-a41c08daecd2572f692ee897da8837a4da089dab5291b6ccfd6d63e9450231fcR392

File: commands.json (hunk 8 above)
Comment (filipecosta90) [reply commands.json:7012]: given this is inherited no change here correct @oranagra @yoav-steinberg?

**Review by yoav-steinberg — APPROVED** [review]
:+1:

File: commands/latency-histogram.md [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+      5# (integer) 16 => (integer) 99968
+      6# (integer) 33 => (integer) 100000
+```
+
+@return
+
+@map-reply: specifically:
+
+The command returns a map where each key is a command name, and each value is a map with the total calls, and an inner map of the histogram time buckets.
+In RESP 2 replies, the maps are converted to arrays.
```
Comment (yoav-steinberg) [inline commands/latency-histogram.md:37]: Until we have RESP3 docs we can't reference map replies.
```suggestion
map-reply: specifically:

The command returns a map where each key is a command name, and each value is a map with the total calls, and an inner map of the histogram time buckets.
In RESP 2 replies, the maps are converted to [arrays](/topics/protocol#array-reply).
```

File: commands.json (hunk 8 above)
Comment (oranagra) [reply commands.json:7012]: yes. it suxx but that's how it is.
the LATENCY command was always tagged as an `admin` command.

File: commands.json [hunk 17]
```diff
@@ -4879,7 +4895,7 @@
     },
     "GEORADIUSBYMEMBER_RO": {
         "summary": "A read-only variant for GEORADIUSBYMEMBER",
-        "since": "3.2.10",
+        "since": "5.0.0",
```
Comment (enjoy-binbin) [inline commands.json:4898]: need to check this: https://github.com/redis/redis/pull/10034

Comment (oranagra) [reply commands.json:4898]: @enjoy-binbin thanks for catching.
apparently @itamarhaber modified (and merged) the contents of commands.json in redis-doc directly, and forgot to merge the change in redis itself.
so it looks as if this PR reverts that chagne.
i now merged https://github.com/redis/redis/pull/10034 so this change will be fixed sooner or later.

File: commands/latency-histogram.md (hunk 7 above)
Comment (zuiderkwast) [reply commands/latency-histogram.md:25]: Candy: redis/redis#10170

Comment (oranagra) [reply commands/latency-histogram.md:25]: thanks.
can you please update this PR with the improved version.
and then merge it ;)

Comment (zuiderkwast) [reply commands/latency-histogram.md:25]:
```suggestion
127.0.0.1:6379> LATENCY HISTOGRAM set
1# "set" =>
   1# "calls" => (integer) 100000
   2# "histogram_usec" =>
      1# (integer) 1 => (integer) 99583
      2# (integer) 2 => (integer) 99852
```
